File: examples/java/integration-test/rest/TestRestClient.java
package com.twilio.rest;

import com.twilio.http.Request;
import com.twilio.http.Response;
import com.twilio.http.TwilioRestClient;

public class TestRestClient extends TwilioRestClient {
    String scheme;

    TestRestClient(final Builder b, final String scheme) {
        super(b);
        this.scheme = scheme;
    }

    @Override
    public Response request(final Request request) {
        StringBuilder sb = new StringBuilder(request.getUrl());
        // manipulate string for mock server url
        sb.replace(0, 5, scheme);
        Request testRequest = new Request(request.getMethod(), sb.toString());
        testRequest.setContentType(request.getContentType());
        // updating the query and post params
        request.getPostParams().forEach((key, value) -> value.forEach(listValue -> testRequest.addPostParam(key, listValue)));
        request.getQueryParams().forEach((key, value) -> value.forEach(listValue -> testRequest.addPostParam(key, listValue)));
        return super.request(testRequest);
    }
}

File: examples/csharp/integrationTest/rest/TestRestClient.cs
using Twilio.Http;
using System;
using System.Text;
using System.Linq;
using System.Collections.Generic;
using Twilio.Clients;
using Twilio.Converters;

namespace examples.csharp.integrationTest.rest
{
    public class TestRestClient : TwilioRestClient, ITwilioRestClient
    {
        String baseURL;

        public TestRestClient( String username, String password, String url) : base(username,password) {
            this.baseURL = url;
        }
        
        public new Response Request(Request request) {
            StringBuilder sb = new StringBuilder(request.Uri.ToString());
            // manipulate string for mock server url
            sb.Remove(0, request.Uri.ToString().IndexOf("twilio.com") + 10);
            sb.Insert(0, baseURL);
            Request testRequest = new Request(request.Method, sb.ToString());
            // updating the query and post params
            request.PostParams.ForEach(val => testRequest.AddPostParam(val.Key, val.Value));
            request.QueryParams.ForEach(val => testRequest.AddPostParam(val.Key, val.Value));
            return base.Request(testRequest);
        }
    }
}